### DISCUSSION TITLE
Is there an upgrade path from v1.3.2?

### DISCUSSION THREAD

**Discussion by MyShoesAreWet** [netbox-community/netbox#7038] [General]
Hello!

I've got a netbox install that is sitting at v1.3.2 and has been for quite a long time. It has quite a lot of data and nobody seems very enthusiastic about re-entering it manually into a fresh install. Is there an upgrade path from the legacy version to the current?  I'll clone the vm and start by trying incremental updates, but if anyone has any advice I'd love to hear it.

**Comment by DanSheps** [comment]
Try in one shot first IMO, make sure you follow the docs, including backing up the database.

**Reply by MyShoesAreWet** [reply]
Cloned and snapshotted at every step. Underlying OS is ubuntu 17. I have my work cut out for me. I think I'll take it as far as I can, then migrate to a new image on a saner OS and import the db.

**Reply by candlerb** [reply]
*In principle* you should be able to build a fresh Netbox 2.11.12 under Ubuntu 20.04.  Once it's working, pgdump and pgrestore the old database to replace the new one, then run `./upgrade.sh` and it should run all the migrations required to update the schema to the new one.

I have occasionally seen some issues in the past which were resolved by going in two steps, but I can't remember off-hand what they were; they may also be triggered by bad data in the database.

If you do the migration and it fails, if you can show the full error report (and backtrace if there is one) from `./upgrade.sh`, it may help identify the issue.  Or you might just be lucky and it all runs cleanly.

**Reply by MyShoesAreWet** [reply]
Cheers for the pointer!

I'm almost done. I followed the surprisingly straightforward steps in this post - https://www.reddit.com/r/networking/comments/ms5p2e/netbox_import_database_on_new_install/ and everything almost works. During the migration script, I get this error:

` Applying circuits.0006_terminations...Traceback (most recent call last):
  File "/opt/netbox/venv/lib/python3.7/site-packages/django/db/backends/utils.py", line 84, in _execute
   return self.cursor.execute(sql, params)
psycopg2.errors.StringDataRightTruncation: value too long for type character varying(1)
`

I don't know if it is because the original DB had some corruption, or if there was some sort of change between the very old and latest versions rendering it incompatible, or if it's just a lot of solar flare activity out there. But the fundamental process seems to be working, we're just down to details now. Thanks!!


EDIT: For anyone with the same trouble, I went into the /circuits/migrations/ folder, found the relevant files (it happened a couple of times) and increased the value from 1 to 10. It should only ever be one char so hopefully I haven't broken anything too badly.
